### PR TITLE
Enable tenv and testifylint rules

### LINKED ISSUE #32842: Enable testifylint and tenv linters in golangci-lint
### Feature Description

[I have contributed this change to Forgejo](https://codeberg.org/forgejo/forgejo/pulls/4535) and I would like to upstream it to Gitea.

Let me know what's the best/recommended way to do this as it did end up large and not pleasant to look through.

I have a few more things that I want to upstream, but I'll have to see first how the codebases differ - maybe they aren't needed.

### Screenshots

_No response_

### PR DESCRIPTION
**Pull request by TheFox0x7**
Enables tenv and testifylint linters
closes: https://github.com/go-gitea/gitea/issues/32842

### CODE REVIEW

**Review by silverwind — APPROVED** [review]

File: services/gitdiff/gitdiff_test.go [hunk 1]
```diff
@@ -643,9 +642,9 @@ func TestGetDiffRangeWithWhitespaceBehavior(t *testing.T) {
 				MaxFiles:           setting.Git.MaxGitDiffFiles,
 				WhitespaceBehavior: behavior,
 			})
-		assert.NoError(t, err, fmt.Sprintf("Error when diff with %s", behavior))
+		assert.NoError(t, err, "Error when diff with %s", behavior)
 		for _, f := range diffs.Files {
-			assert.True(t, len(f.Sections) > 0, fmt.Sprintf("%s should have sections", f.Name))
+			assert.Positive(t, f.Sections, "%s should have sections", f.Name)
```
Comment (wxiaoguang) [inline services/gitdiff/gitdiff_test.go]: Like this?

```suggestion
			assert.NotEmpty(t, f.Sections, "%s should have sections", f.Name)
```

(I am not sure whether "Positive" works for a slice)

File: tests/integration/mirror_push_test.go [hunk 2]
```diff
@@ -78,7 +78,7 @@ func testMirrorPush(t *testing.T, u *url.URL) {
 	assert.True(t, doRemovePushMirror(t, session, user.Name, srcRepo.Name, mirrors[0].ID))
 	mirrors, _, err = repo_model.GetPushMirrorsByRepoID(db.DefaultContext, srcRepo.ID, db.ListOptions{})
 	assert.NoError(t, err)
-	assert.Len(t, mirrors, 0)
+	assert.Empty(t, mirrors, 0)
```
Comment (silverwind) [inline tests/integration/mirror_push_test.go]:
```suggestion
	assert.Empty(t, mirrors)
```

File: tests/integration/api_repo_lfs_locks_test.go [hunk 3]
```diff
@@ -176,6 +176,6 @@ func TestAPILFSLocksLogged(t *testing.T) {
 		resp := session.MakeRequest(t, req, http.StatusOK)
 		var lfsLocks api.LFSLockList
 		DecodeJSON(t, resp, &lfsLocks)
-		assert.Len(t, lfsLocks.Locks, 0)
+		assert.Empty(t, lfsLocks.Locks, 0)
```
Comment (silverwind) [inline tests/integration/api_repo_lfs_locks_test.go]:
```suggestion
		assert.Empty(t, lfsLocks.Locks)
```

File: tests/integration/mirror_push_test.go (hunk 2 above)
Comment (wxiaoguang) [inline tests/integration/mirror_push_test.go]:
```suggestion
	assert.Empty(t, mirrors)
```

File: tests/integration/api_repo_lfs_locks_test.go (hunk 3 above)
Comment (wxiaoguang) [inline tests/integration/api_repo_lfs_locks_test.go]:
```suggestion
		assert.Empty(t, lfsLocks.Locks)
```

File: tests/integration/api_notification_test.go [hunk 4]
```diff
@@ -154,7 +154,7 @@ func TestAPINotification(t *testing.T) {
 		AddTokenAuth(token)
 	resp = MakeRequest(t, req, http.StatusOK)
 	DecodeJSON(t, resp, &newStruct)
-	assert.True(t, newStruct.New == 0)
+	assert.Equal(t, 0, newStruct.New)
```
Comment (wxiaoguang) [inline tests/integration/api_notification_test.go]:
```suggestion
	assert.EqualValues(t, 0, newStruct.New)
```

`newStruct.New` seems to be a `int64`

Comment (wxiaoguang) [reply tests/integration/api_notification_test.go]: (or `assert.Zero`)

```suggestion
	assert.Zero(t, newStruct.New)
```

**Review by wxiaoguang — APPROVED** [review]

**Review by silverwind — APPROVED** [review]
lgtm if it passes

**Pull request merged**